ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/context/KsqlRestContextProvider.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.rest.server.context;
+
+import java.util.Collections;
+
+public final class KsqlRestContextProvider {
+  private static final ThreadLocal<KsqlRestContext> restContextThreadLocal = new ThreadLocal<>();
```

Comment:
This seems like a hacky... we shouldn't be using ThreadLocal as an API to communicate information (aside from the fact that ThreadLocal can be a dangerous abstraction). This also makes the APIs very confusing... I have no idea what's going on.

Reply:
I agree. Is there any guarantee filters and resources are called on the same thread?